### ISSUE TITLE
Syntax highlighting for Python/SageTeX doesn't work until Vim is exited and reopened?

### ISSUE THREAD

**Issue by user202729** [lervag/vimtex#2635]
Let's say, I have something like

```tex
\documentclass{article}
\begin{document}

\begin{pycode}
print(1)
\end{pycode}

The answer is \sagestr{print([i for i in range(5)])}.

\end{document}
```

the syntax highlighting for the Python inside does not work...

Then I add `\usepackage{pythontex}` and `\usepackage{sagetex}`, it still doesn't get syntax highlighted.

Reloading the document with `:e%` doesn't work, as well as `:bdelete` and opening the file again. Only completely exiting Vim and reopen it works.

How to workaround this? (I assume avoid loading syntax file when the package is not loaded is for performance reason, but the behavior is a bit unexpected?)

**Comment by lervag** [comment]
Yes, this is a known limitation. Alternatives/Workarounds:

* When you document is compiled, it should trigger a reload of syntax package extensions. This should work also in the example you bring; if you were to compile the document after you add `\usepackage...`, VimTeX should load the relevant syntax packages.
* You can use `\lx` or `:VimtexReload` to reload VimTeX entirely. This is often a good alternative.
* You can use `g:vimtex_syntax_packages` to specify that an extension should always load. See the docs, I believe they should explain this well.

**Comment by lervag** [comment]
> I assume avoid loading syntax file when the package is not loaded is for performance reason, but the behavior is a bit unexpected?

Yes, precisely. Syntax stuff is one of the more expensive things in Vim/neovim; I believe (but I've actually never tested this properly) that the lazy loading of syntax packages gives a noticable impact on efficiency.

I'm aware it can be unexpected, but I've tried to add useful workarounds and options and I've tried to explain these things in the docs. See `:help vimtex-syntax`.
